File: src/main/java/trop/TROPItemRingVilia.java
package trop;

import net.minecraft.entity.Entity;
import net.minecraft.entity.player.EntityPlayer;
import net.minecraft.init.MobEffects;
import net.minecraft.item.*;
import net.minecraft.potion.PotionEffect;
import net.minecraft.util.*;
import net.minecraft.world.World;

public class TROPItemRingVilia extends Item {
	public TROPItemRingVilia(Properties prop) {
		super(prop);
	}

	@Override
	public void inventoryTick(ItemStack itemstack, World world, Entity entity, int par4, boolean par5) {
		if (entity instanceof EntityPlayer) {
			((EntityPlayer) entity).addPotionEffect(new PotionEffect(MobEffects.SPEED, 20, 2));
			((EntityPlayer) entity).addPotionEffect(new PotionEffect(MobEffects.JUMP_BOOST, 20, 2));
		}
	}

	@Override
	public ActionResult<ItemStack> onItemRightClick(World world, EntityPlayer entity, EnumHand hand) {
		entity.addPotionEffect(new PotionEffect(MobEffects.REGENERATION, 3600, 2));
		entity.addPotionEffect(new PotionEffect(MobEffects.HEALTH_BOOST, 3800, 2));
		return super.onItemRightClick(world, entity, hand);
	}
}

File: kotlin/src/main/java/trop/TROPItemRingVilia.kt
package trop

import net.minecraft.world.InteractionHand
import net.minecraft.world.InteractionResultHolder
import net.minecraft.world.effect.MobEffectInstance
import net.minecraft.world.effect.MobEffects
import net.minecraft.world.entity.Entity
import net.minecraft.world.entity.player.Player
import net.minecraft.world.item.ItemStack
import net.minecraft.world.level.Level

class TROPItemRingVilia : TROPItemRingBase() {
    override fun inventoryTick(itemstack: ItemStack, world: Level, entity: Entity, par4: Int, par5: Boolean) {
        if (entity is Player) {
            entity.addEffect(MobEffectInstance(MobEffects.MOVEMENT_SPEED, 20, 2))
            entity.addEffect(MobEffectInstance(MobEffects.JUMP, 20, 2))
        }
    }

    override fun use(world: Level, entity: Player, hand: InteractionHand): InteractionResultHolder<ItemStack> {
        entity.addEffect(MobEffectInstance(MobEffects.REGENERATION, 3600, 2))
        entity.addEffect(MobEffectInstance(MobEffects.HEALTH_BOOST, 3800, 2))
        return super.use(world, entity, hand)
    }
}
